### PR TITLE
User/bulelani/cb 1824/add test for subarray sensor lookup

### PR DESCRIPTION
**Pull request by bxaia**
@fjoubert / @ajoubertza please review

### CODE REVIEW

File: katportalclient/test/test_client.py [hunk 1]
```diff
@@ -1331,6 +1333,50 @@ def test_modify_userlog(self):
...
+
+        self.mock_http_async_client().fetch.side_effect = mock_async_fetcher(
+                valid_response='{"result":"cbf_3.device-status"}',
+                invalid_response=['error'],
+                starts_with=lookup_base_url,
+                contains=sensor_name_filter)
+        sensor = yield self._portal_client.sensor_subarray_lookup('cbf', sensor_name_filter, True)
+        self.assertTrue(sensor == expected_sensor_name)
+
+
```
Comment (ajoubertza) [inline katportalclient/test/test_client.py]: There is one extra blank here.  Normally we want 1 blank line between class methods, and 2 blank lines between functions.

File: katportalclient/test/test_client.py [hunk 2]
```diff
@@ -1331,6 +1333,50 @@ def test_modify_userlog(self):
             userlog_to_modify['tags'] = 'random nonsense'
             userlog = yield self._portal_client.modify_userlog(userlog_to_modify)
 
+    @gen_test
+    def test_sensor_subarray_lookup(self):
+        """Test sensor subarray lookup is correctly extracted."""
+        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/cbf/device_status/0'
```
Comment (ajoubertza) [inline katportalclient/test/test_client.py]: This line is a little bit too long.  Please keep <= 90 chars, as per our coding guidelines.
Same for the line in the other tests.

File: katportalclient/test/test_client.py [hunk 3]
```diff
@@ -1331,6 +1333,50 @@ def test_modify_userlog(self):
...
+        self.assertTrue(sensor == expected_sensor_name)
+
+
+    @gen_test
+    def test_sensor_subarray_invalid_sensor_lookup(self):
+        """Test that sensor subarray lookup can correctly handle an invalid sensor name."""
+        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/anc/device_status/0'
+        sensor_name_filter = 'device_status'
+        self.mock_http_async_client().fetch.side_effect = mock_async_fetcher(
+                valid_response='{"error":"SensorLookupError: Could not lookup the sensor on component. Could not determine component."}',
```
Comment (ajoubertza) [inline katportalclient/test/test_client.py]: Here's another line that is a bit long.  E.g.
```
    valid_response='{"error": "SensorLookupError: Could not lookup the sensor on component. '
                   'Could not determine component."}',
```

File: katportalclient/test/test_client.py (hunk 2 above)
Comment (fjoubert) [reply katportalclient/test/test_client.py]: Agreed. Lines like these you can format like:
```
lookup_base_url = '{}/3/sensor-lookup/cbf/device_status/0'.format(
    self._portal_client.sitemap['subarray'])
```
or even
```
lookup_base_url = (self._portal_client.sitemap['subarray'] +
                   '/3/sensor-lookup/cbf/device_status/0')
```

Comment (bxaia) [reply katportalclient/test/test_client.py]: I've addressed the suggested changes on the next commit

File: katportalclient/test/test_client.py [hunk 4]
```diff
@@ -1336,7 +1336,8 @@ def test_modify_userlog(self):
     @gen_test
     def test_sensor_subarray_lookup(self):
         """Test sensor subarray lookup is correctly extracted."""
-        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/cbf/device_status/0'
+        lookup_base_url = (self._portal_client.sitemap['subarray'] + 
+        '/3/sensor-lookup/cbf/device_status/0')
```
Comment (fjoubert) [inline katportalclient/test/test_client.py]: This indentation is should be 4 spaces in, like so:
```
lookup_base_url = (self._portal_client.sitemap['subarray'] +
    '/3/sensor-lookup/cbf/device_status/0')
```

File: katportalclient/test/test_client.py [hunk 5]
```diff
@@ -1345,13 +1346,15 @@ def test_sensor_subarray_lookup(self):
                 invalid_response=['error'],
                 starts_with=lookup_base_url,
                 contains=sensor_name_filter)
-        sensor = yield self._portal_client.sensor_subarray_lookup('cbf', sensor_name_filter, False)
+        sensor = yield self._portal_client.sensor_subarray_lookup(
+                'cbf', sensor_name_filter, False)
```
Comment (fjoubert) [inline katportalclient/test/test_client.py]: Too many spaces here 😄 
```
sensor = yield self._portal_client.sensor_subarray_lookup(
    'cbf', sensor_name_filter, False)
````

File: katportalclient/test/test_client.py [hunk 6]
```diff
@@ -1345,13 +1346,15 @@ def test_sensor_subarray_lookup(self):
...
         self.assertTrue(sensor == expected_sensor_name)
 
     @gen_test
     def test_sensor_subarray_katcp_name_lookup(self):
         """Test sensor subarray lookup returns the correct katcp name."""
-        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/cbf/device-status/1'
+        lookup_base_url = (self._portal_client.sitemap['subarray'] + 
+        '/3/sensor-lookup/cbf/device-status/1')
```
Comment (fjoubert) [inline katportalclient/test/test_client.py]: Ditto 4 spaces indent here

File: katportalclient/test/test_client.py [hunk 7]
```diff
@@ -1360,23 +1363,25 @@ def test_sensor_subarray_katcp_name_lookup(self):
                 invalid_response=['error'],
                 starts_with=lookup_base_url,
                 contains=sensor_name_filter)
-        sensor = yield self._portal_client.sensor_subarray_lookup('cbf', sensor_name_filter, True)
+        sensor = yield self._portal_client.sensor_subarray_lookup(
+                'cbf', sensor_name_filter, True)
```
Comment (fjoubert) [inline katportalclient/test/test_client.py]: Ditto too many spaces here

File: katportalclient/test/test_client.py [hunk 8]
```diff
@@ -1360,23 +1363,25 @@ def test_sensor_subarray_katcp_name_lookup(self):
...
+                valid_response='{"error":"SensorLookupError: Could not lookup the sensor '
+                'on component. Could not determine component."}',
                 invalid_response='[]',
                 starts_with=lookup_base_url,
                 contains=sensor_name_filter)
         with self.assertRaises(SensorLookupError):
-            yield self._portal_client.sensor_subarray_lookup('anc', sensor_name_filter, False)
-
+            yield self._portal_client.sensor_subarray_lookup(
+                    'anc', sensor_name_filter, False)
```
Comment (fjoubert) [inline katportalclient/test/test_client.py]: Too many spaces for indentation here, rather do:
```
yield self._portal_client.sensor_subarray_lookup(
    'anc', sensor_name_filter, False)
```

File: katportalclient/test/test_client.py [hunk 9]
```diff
@@ -1360,23 +1363,25 @@ def test_sensor_subarray_katcp_name_lookup(self):
...
+                'cbf', sensor_name_filter, True)
         self.assertTrue(sensor == expected_sensor_name)
 
-
     @gen_test
     def test_sensor_subarray_invalid_sensor_lookup(self):
         """Test that sensor subarray lookup can correctly handle an invalid sensor name."""
-        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/anc/device_status/0'
+        lookup_base_url = (self._portal_client.sitemap['subarray'] + 
+        '/3/sensor-lookup/anc/device_status/0')
```
Comment (fjoubert) [inline katportalclient/test/test_client.py]: Ditto spaces

File: katportalclient/test/test_client.py [hunk 10]
```diff
@@ -1360,23 +1363,25 @@ def test_sensor_subarray_katcp_name_lookup(self):
...
         self.mock_http_async_client().fetch.side_effect = mock_async_fetcher(
-                valid_response='{"error":"SensorLookupError: Could not lookup the sensor on component. Could not determine component."}',
+                valid_response='{"error":"SensorLookupError: Could not lookup the sensor '
+                'on component. Could not determine component."}',
                 invalid_response='[]',
                 starts_with=lookup_base_url,
                 contains=sensor_name_filter)
         with self.assertRaises(SensorLookupError):
-            yield self._portal_client.sensor_subarray_lookup('anc', sensor_name_filter, False)
-
```
Comment (ajoubertza) [inline katportalclient/test/test_client.py]: Removing this blank line wasn't necessary, as we have a function defined next (not a method).  We need two lines before a function.  These kind of rules are hard to remember, so I like my editor to remind me.  If you have an editor configured to show you flake8 warnings, it is really easy to spot.  E.g., using the Atom editor with the flake8 linter plugin:

image

Similar things can be done in vi.

**Pull request merged**